File: core/src/com/alta_v2/physics/task/rotatePlayer/RotateNpcChecker.java
package com.alta_v2.physics.task.rotatePlayer;

import com.alta_v2.physics.executionContext.TiledMapEngineContext;
import com.alta_v2.physics.executionContext.reserveData.ReservableActor;
import com.alta_v2.physics.task.ExecutionChecker;
import com.alta_v2.rendering.actors.PersonView;
import lombok.RequiredArgsConstructor;

/**
 * Provides the checker that indicates when NPC can be rotate.
 */
@RequiredArgsConstructor
public class RotateNpcChecker implements ExecutionChecker {

    private final String npcId;
    private final PersonView currentView;

    @Override
    public boolean canTaskBeExecuted(TiledMapEngineContext context) {
        ReservableActor npc = context.getNpcMap().get(this.npcId);
        return !npc.getView().isReserved() && npc.getView().getValue() != this.currentView;
    }
}

File: core/src/com/alta_v2/physics/task/rotatePlayer/RotateNpcChecker.kt
package com.alta_v2.physics.task.rotatePlayer

import com.alta_v2.physics.executionContext.TiledMapEngineContext
import com.alta_v2.physics.task.ExecutionChecker
import com.alta_v2.rendering.tiledMapScreen.actors.PersonView

/**
 * Provides the checker that indicates when NPC can be rotate.
 */
class RotateNpcChecker(private val npcId: Int, private val currentView: PersonView) : ExecutionChecker {
    override fun canTaskBeExecuted(context: TiledMapEngineContext): Boolean {
        val npc = context.npcMap[npcId]
        return !npc!!.view.isReserved() && npc.view.getValue() != currentView
    }
}